spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/CassandraConnectionFactory.scala

```diff
@@ -77,9 +77,33 @@ object DefaultConnectionFactory extends CassandraConnectionFactory {
         } finally {
           IOUtils.closeQuietly(trustStoreFile)
         }
-
+        val kmf = if(conf.clientAuthEnabled){
+            conf.keyStorePath map {
+              case path ⇒
+                val keyStoreFile = new FileInputStream(path)
+                val kmf = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+                val keyStore = KeyStore.getInstance(conf.keyStoreType)
+                conf.keyStorePassword match {
+                  case None ⇒ {
+                    keyStore.load(keyStoreFile, null)
+                    kmf.init(keyStore, null)
+                  }
+                  case Some(password) ⇒ {
+                    keyStore.load(keyStoreFile, password.toCharArray)
+                    kmf.init(keyStore, password.toCharArray)
+                  }
+                }
+                kmf
+            }
+          }else{
+            None
+          }
         val context = SSLContext.getInstance(conf.protocol)
-        context.init(null, tmf.getTrustManagers, new SecureRandom)
+        kmf match {
```

Comment:
We should either support combination when keystore is specified but truststore is not or throw exception in such a case.